fe/fe-core/src/main/java/com/starrocks/sql/plan/PlanFragmentBuilder.java

```diff
@@ -1644,14 +1670,17 @@ private PlanFragment visitPhysicalJoin(PlanFragment leftFragment, PlanFragment r
                             rightFragmentPlanRoot.canDoReplicatedJoin()) {
                         distributionMode = JoinNode.DistributionMode.REPLICATED;
                     } else if (isShuffleJoin(optExpr)) {
+                        probePartitionByExprs = getHashDistributionSpecPartitionByExprsOfChild(optExpr, context, 0);
                         distributionMode = JoinNode.DistributionMode.SHUFFLE_HASH_BUCKET;
                     } else {
                         Preconditions.checkState(false, "Must be replicate join or colocate join");
                         distributionMode = JoinNode.DistributionMode.COLOCATE;
                     }
                 } else if (isShuffleJoin(optExpr)) {
+                    probePartitionByExprs = getHashDistributionSpecPartitionByExprsOfChild(optExpr, context, 0);
```

Comment:
At present, are colcoated HJ and replicated HJ not be taken into consideration? if so, please add a TODO notice.